### PR TITLE
[release/v2.20] adjust resources and probes for konnectivity proxy sidecar (#9911)

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What does this PR do / Why do we need it**:
- Increases memory limits for konnectivity-proxy sidecar. 
- Allows overriding the limits via components override in the cluster-resource. 
- Adjusts readiness  probe timings.

**Does this PR close any issues?**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix probes, resources and allow overriding resource requests/limits for Konnectivity proxy via components override in the cluster resource
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/retest

Issue with MLA job is fixed now.

File: pkg/apis/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -481,12 +481,22 @@ type MLASettings struct {
...
+	// Etcd configures the etcd ring used to store Kubernetes data.
+	Etcd EtcdStatefulSetSettings `json:"etcd"`
+	// Prometheus configures the Prometheus instance deployed into the cluster control plane.
+	Prometheus StatefulSetSettings `json:"prometheus"`
+	// NodePortProxyEnvoy configures the per-cluster nodeport-proxy-envoy that is deployed if
+	// the `LoadBalancer` expose strategy is used. This is not effective if a different expose
+	// strategy is configured.
+	NodePortProxyEnvoy NodeportProxyComponent `json:"nodePortProxyEnvoy"`
+	// KonnectivityProxy configures resources limits/requests for konnectivity-server sidecar.
+	KonnectivityProxy KonnectvityProxySettings `json:"konnectivityProxy"`
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/cluster.go]: for backward compatibility this field should be optional (i.e. have an omitempty).

**Pull request merged**